src/sql/workbench/contrib/query/browser/fileQueryResultsWriter.ts

```diff
@@ -0,0 +1,341 @@
+import QueryRunner, { QueryGridDataProvider } from 'sql/workbench/services/query/common/queryRunner';
+import { asArray } from 'vs/base/common/arrays';
+import { CancellationToken } from 'vs/base/common/cancellation';
+import { Disposable } from 'vs/base/common/lifecycle';
+import { isArray } from 'vs/base/common/types';
+import { isWindows } from 'vs/base/common/platform';
+import { formatDocumentWithSelectedProvider, FormattingMode } from 'vs/editor/contrib/format/format';
+import * as nls from 'vs/nls';
+import { IConfigurationService } from 'vs/platform/configuration/common/configuration';
+import { IInstantiationService } from 'vs/platform/instantiation/common/instantiation';
+import { ILogService } from 'vs/platform/log/common/log';
+import { Progress } from 'vs/platform/progress/common/progress';
+import { IEditorService } from 'vs/workbench/services/editor/common/editorService';
+import { IUntitledTextEditorService } from 'vs/workbench/services/untitled/common/untitledTextEditorService';
+
+export class FileQueryResultsWriter implements IQueryResultsWriter {
+	private messages: Array<IQueryMessage> = [];
+	private formattedQueryResults: Array<string> = [];
+	private tables: Array<Table<any>> = [];
+	private numQueriesToFormat: number = 0;
+	private queryContainsError: boolean = false;
+	private closingMessageIncluded: boolean = false;
+	private hasCreatedResultsFile: boolean = false;
+
+	private readonly startedExecutingQueryMessage: string;
+	private readonly totalExecutionTimeMessage: string;
+	private readonly rowAffectedMessage: string;
+	private readonly rowsAffectedMessage: string;
+	private readonly newLineEscapeSequence: string;
+
+	constructor(
+		private runner: QueryRunner,
+		@IInstantiationService private instantiationService: IInstantiationService,
+		@IUntitledTextEditorService private readonly untitledEditorService: IUntitledTextEditorService,
+		@IEditorService private readonly editorService: IEditorService,
+		@ILogService private readonly logService: ILogService,
+		@IConfigurationService private configurationService: IConfigurationService
+	) {
+		this.startedExecutingQueryMessage = nls.localize('query.message.startingExecutionQuery', 'Started executing query');
+		this.totalExecutionTimeMessage = nls.localize('query.message.totalExecutionTime', 'Total execution time');
+		this.rowAffectedMessage = nls.localize('query.message.rowAffected', 'row affected');
+		this.rowsAffectedMessage = nls.localize('query.message.rowsAffected', 'rows affected');
+		this.newLineEscapeSequence = isWindows ? '\r\n' : '\n';
+	}
+
+	public onQueryStart() {
+		this.reset();
+	}
+
+	public onResultSet(resultSet: ResultSetSummary | ResultSetSummary[]) {
+		this.numQueriesToFormat++;
+		let resultsToAdd: ResultSetSummary[];
+		if (!Array.isArray(resultSet)) {
+			resultsToAdd = [resultSet];
+		} else {
+			resultsToAdd = asArray(resultSet);
```

Comment:
I meant you can simplify this entire thing - so 

```
let resultsToAdd: ResultSetSummary[];
if (!Array.isArray(resultSet)) {
	resultsToAdd = [resultSet];
} else {
	resultsToAdd = asArray(resultSet);
}
```

becomes just

`let resultsToAdd = asArray(resultSet);`